### PR TITLE
Sync the "master" branch to "main"

### PR DESCRIPTION
**Pull request by dmikurube**

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    branches:
+      - "master"
+jobs:
+  diff:
+    runs-on: ubuntu-latest
+    outputs:
+      hash-master: ${{ steps.hash-master.outputs.hash-master }}
+      hash-main: ${{ steps.hash-main.outputs.hash-main }}
+    steps:
+    - uses: actions/checkout@v3
```
Comment (hiroyuki-sato) [inline .github/workflows/main.yml]: What do you think to use `actions/checkout@v4`?

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+      run: |
+        hash_main=$( git rev-parse origin/main )
+        echo "$hash_main"
+        echo "hash-main=$hash_main" >> $GITHUB_OUTPUT
+  sync:
+    needs: diff
+    if: needs.diff.outputs.hash-master != needs.diff.outputs.hash-main
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
```
Comment (hiroyuki-sato) [inline .github/workflows/main.yml]: ditto

**Review by hiroyuki-sato — APPROVED** [review]
LTGM👍 (Some minor comments)

File: .github/workflows/main.yml (hunk 1 above)
Comment (dmikurube) [reply .github/workflows/main.yml]: Good catch. Thanks!

**Pull request merged**